packages/react-graphql/src/hooks/types.ts

```diff
@@ -1,5 +1,23 @@
 import ApolloClient from 'apollo-client';
-import {MutationOptions, FetchResult, OperationVariables} from 'react-apollo';
+import {
+  QueryOpts,
+  QueryResult,
+  MutationOptions,
+  FetchResult,
+  OperationVariables,
+} from 'react-apollo';
+import {Omit} from '@shopify/useful-types';
+
+export interface QueryHookOptions<Variables = OperationVariables>
+  extends Omit<QueryOpts<Variables>, 'partialRefetch'> {
+  skip?: boolean;
+}
+
+export interface QueryHookResult<Data, Variables>
+  extends Omit<QueryResult<Data, Variables>, 'networkStatus' | 'variables'> {
+  networkStatus: QueryResult<Data, Variables>['networkStatus'] | undefined;
+  variables: QueryResult<Data, Variables>['variables'] | undefined;
```

Comment:
We can technically offer up better typings for some of these because we can check if the variables have any required keys or not, but for now this is fine 👍